src/Symfony/Component/HttpKernel/Profiler/Profiler.php

```diff
@@ -26,21 +26,19 @@
  */
 class Profiler implements ResetInterface
 {
-    private ProfilerStorageInterface $storage;
-
     /**
      * @var DataCollectorInterface[]
      */
     private array $collectors = [];
 
-    private ?LoggerInterface $logger;
     private bool $initiallyEnabled = true;
     private bool $enabled = true;
 
-    public function __construct(ProfilerStorageInterface $storage, ?LoggerInterface $logger = null, bool $enable = true)
-    {
-        $this->storage = $storage;
-        $this->logger = $logger;
+    public function __construct(
+        private ProfilerStorageInterface $storage,
+        private ?LoggerInterface $logger = null,
+        bool $enable = true,
```

Comment:
CPP is possible for this one also

```suggestion
        private bool $enabled = true,
```

Reply:
I thought about that, but that would mean we would change the parameter name from `$enable` to `$enabled`. Constructors with optional parameters are likely to be invoked with named arguments. Is changing the parameter name acceptable in this case?

Same question for `ExceptionEvent` where the parameter `$e` is stored inside a property `$throwable`.

Reply:
Our BC policy says that argument names are covered by BC only for constructor of attributes and annotations.

Reply:
Alternatively, you can rename the private property to `$enable`.

Reply:
Right. But then again, the class is not final and `$enabled` has a property initializer again. 😢

Reply:
enabled is private so this doesn't matter, right?
It'd do the renaming. The profiler is not the thing I'd expected anyone to call with named arguments.

Reply:
Okay, what about the property initializer? Shall I ignore that as well for the `Profiler` class?

Reply:
> what about the property initializer?

sorry I'm not sure what you mean.

Reply:
https://github.com/symfony/symfony/blob/749ad6e7a27e44af09803e73c4f9090c3f7ddc8e/src/Symfony/Component/HttpKernel/Profiler/Profiler.php#L38

Using CPP for this property would leave it uninitialized if the constructor is skipped.

Reply:
OK, this won't matter in practice.

Reply:
> OK, this won't matter in practice.

🤞🏻